sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowPythonRunner.scala

```diff
@@ -48,6 +49,12 @@ class ArrowPythonRunner(
   extends BasePythonRunner[Iterator[InternalRow], ColumnarBatch](
     funcs, evalType, argOffsets) {
 
+  override val bufferSize: Int = SQLConf.get.pandasUDFBufferSize
+  require(
+    bufferSize >= 4,
```

Comment:
Is this cap here not in SQL config definition because fallback of `BUFFER_SIZE`?

Reply:
Yea, I think it actually should be in `BUFFER_SIZE` ... I wonder in what case people will set the buffer lower than 4 bytes.

Currently, Python workers don't work when the buffer size is lower than 4 (`struct` package doesn't allow that) so I only applied this condition to Pandas UDF execution for now.

Actually, if it makes sense to all of us, I'd like to bring this condition to `spark.buffer.size` itself.

Reply:
cc @vanzin too. Do you think it makes sense to just add the condition to `spark.buffer.size` that the buffer should be higher than 4 bytes? If we're not sure yet, I'll just keep this condition here.

Reply:
Would it be possible to just check internally if the buffer size is 0 and less than 4, then don't create a BufferedWriter in Scala at all? If someone really wants a buffer size that low, I think it makes sense to just turn off buffering.